datafusion/physical-expr/src/conditional_expressions.rs

```diff
@@ -0,0 +1,80 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+use std::sync::Arc;
+
+use arrow::array::{new_null_array, Array};
+use arrow::compute;
+use arrow::compute::kernels::zip::zip;
+use arrow::datatypes::DataType;
+
+use datafusion_common::{DataFusionError, Result, ScalarValue};
+use datafusion_expr::ColumnarValue;
+
+pub fn coalesce(args: &[ColumnarValue]) -> Result<ColumnarValue> {
+    // do not accept 0 arguments.
+    if args.is_empty() {
+        return Err(DataFusionError::Internal(format!(
+            "coalesce was called with {} arguments. It requires at least 1.",
+            args.len()
+        )));
+    }
+
+    let size = match args[0] {
+        ColumnarValue::Array(ref a) => a.len(),
+        ColumnarValue::Scalar(ref _s) => 1,
+    };
+    let mut res = new_null_array(&args[0].data_type(), size);
+
+    for column_value in args {
+        for i in 0..size {
+            match column_value {
+                ColumnarValue::Array(array_ref) => {
+                    let bool_arr = compute::is_not_null(array_ref)?;
+                    res = zip(&bool_arr, array_ref, &res)?;
+                }
+                ColumnarValue::Scalar(scalar) => {
+                    if !scalar.is_null() {
+                        // TODO: Figure out how to set value at index
```

Comment:
Hi @alamb, I need your help here. I couldn't figure out how to set value to particular index of `arrow::Array`

Reply:
I think you may want to call `scalar. to_array_of_size(size)` here to create the appropriate array:

https://github.com/alamb/arrow-datafusion/blob/41b4e49/datafusion/common/src/scalar.rs#L1075-L1076